src/Symfony/Component/DependencyInjection/Tests/ContainerBuilderTest.php

```diff
@@ -1316,6 +1316,47 @@ public function testNoClassFromNsSeparatorId()
         $container->compile();
     }
 
+    public function testGetThrownServiceNotFoundExceptionWithCorrectServiceId()
+    {
+        $this->expectException(ServiceNotFoundException::class);
+        $this->expectExceptionMessage('The service "child_service" has a dependency on a non-existent service "non_existent_service".');
+
+        $container = new ContainerBuilder();
+        $container->register('child_service', \stdClass::class)
+            ->setPublic(false)
+            ->addArgument([
+                'non_existent' => new Reference('non_existent_service'),
+            ])
+        ;
+        $container->register('parent_service', \stdClass::class)
+            ->setPublic(true)
+            ->addArgument([
+                'child_service' => new Reference('child_service'),
+            ])
+        ;
+
+        $container->compile();
+    }
+
+    public function testUnusedServiceRemovedByPassAndServiceNotFoundExceptionWasNotThrown()
+    {
+        $container = new ContainerBuilder();
+        $container->register('service', \stdClass::class)
+            ->setPublic(false)
+            ->addArgument([
+                'non_existent_service' => new Reference('non_existent_service'),
+            ])
+        ;
+
+        try {
+            $container->compile();
+        } catch (ServiceNotFoundException $e) {
+            $this->fail('Should not be thrown');
+        }
+
+        $this->assertTrue(true);
```

Comment:
you can use either `$this->expectNotToPerformAssertions()` or `$this->addToAssertionCount(1)`

Reply:
done